api/swagger.yaml

```diff
@@ -181,132 +177,168 @@ paths:
               schema:
                 $ref: "#/components/schemas/Reports"
         "403":
-          description: Not authorized
+          description: Please login again
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+
 components:
   schemas:
     Command:
       required:
       - type
-      type: object
       properties:
         type:
           type: string
-      discriminator:
-        propertyName: Type_
-      oneOf:
-      - $ref: '#/components/schemas/MoveCommand'
-      - $ref: '#/components/schemas/StopCommand'
-      - $ref: '#/components/schemas/ConstructCommand'
-      - $ref: '#/components/schemas/DecommissionCommand'
-      - $ref: '#/components/schemas/AttackCommand'
-      - $ref: '#/components/schemas/TradeCommand'
-      - $ref: '#/components/schemas/RenameCommand'
+        target:
+          type: integer
+          format: int64
+          nullable: true
+        destination:
+          $ref: "#/components/schemas/Destination"
+        resource:
+          type: integer
+          format: int64
+          nullable: true
+        amount:
+          type: integer
+          format: int64
+          nullable: true
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
+        name:
+          type: string
+          nullable: true
+
     MoveCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - where
-        type: object
-        properties:
-          destination:
-            $ref: "#/components/schemas/Destination"
+      type: object
+      properties:
+        type:
+          type: string
+        destination:
+          $ref: "#/components/schemas/Destination"
+      required:
+        - type
+        - destination
+
     StopCommand:
-      $ref: '#/components/schemas/Command'
+      type: object
 
     ConstructCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - class
-        type: object
-        properties:
-          class:
-            type: string
-          name:
-            type: string
+      type: object
+      properties:
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
+        type:
+          type: string
+      required:
+        - type
+        - ship-class
 
     DecommissionCommand:
-      $ref: '#/components/schemas/Command'
+      type: object
+      properties: 
+        type:
+          type: string
+      required:
+        - type
 
     RenameCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
+      type: object
+      properties:
+        type:
```

Comment:
what is type?

Reply:
It's the command type that must be present for all commands. It's there to make sure it's not missing. Perhaps it could be only left in Command object.

Reply:
> Perhaps it could be only left in Command object.

yes please

Reply:
done

Reply:
Unrersolving - how are now clients supposed to use the specific commands objects, if they are missing the type?

Reply:
the type of the command is in the name of type :D
AttackCommand is attack, right?

Reply:
discussed "offline" - the type serves the purpose of using the structure as a standalone command to be filled in the commands map by clients. Let's keep the type for now and revisit later if needed.